Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -105,26 +105,26 @@ public void Start()
 
         #region Client
 
-        void RegisterClientMessages(INetworkPlayer player)
+        void RegisterClientMessages()
         {
-            player.RegisterHandler<SceneMessage>(ClientSceneMessage);
+            Client.MessageHandler.RegisterHandler<SceneMessage>(ClientSceneMessage);
             if (!Client.IsLocalClient)
             {
-                player.RegisterHandler<SceneReadyMessage>(ClientSceneReadyMessage);
-                player.RegisterHandler<NotReadyMessage>(ClientNotReadyMessage);
+                Client.MessageHandler.RegisterHandler<SceneReadyMessage>(ClientSceneReadyMessage);
+                Client.MessageHandler.RegisterHandler<NotReadyMessage>(ClientNotReadyMessage);
             }
         }
 
-        void OnClientAuthenticated(INetworkPlayer player)
+        void OnDestroy()
         {
-            logger.Log("NetworkSceneManager.OnClientAuthenticated");
-            RegisterClientMessages(player);
+            if (Client != null)
+                Client.Authenticated?.RemoveListener(OnClientConnected);
```

Comment:
You use Connected higher up to register but Authenticated here to unregister.

Reply:
good catch, I'll fix this